hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobExecutionService.java

```diff
@@ -327,6 +329,24 @@ public void completeExecution(long executionId, Throwable error) {
         }
     }
 
+    public void updateMetrics(Map<String, Long> metrics) {
+        Map<ExecutionContext, Map<String, Long>> metricsOfExecutions = Collections.emptyMap();
+
+        for (Entry<String, Long> entry : metrics.entrySet()) {
+            String metricName = entry.getKey();
+            Long executionId = JobMetricsUtil.getExecutionIdFromMetricName(metricName);
+            if (executionId != null) {
+                ExecutionContext executionContext = executionContexts.get(executionId);
+                if (executionContext != null) {
+                    Long metricValue = entry.getValue();
+                    metricsOfExecutions = putIntoMultiMap(metricsOfExecutions, executionContext, metricName, metricValue);
```

Comment:
why do we need the complexity of this method? you can just use `map.compute()` instead.

Reply:
Yeah, I just thought to create a smaller number of full blown `Map` instances (ie. `Collections.singletonMap` instead of `HashMap`), but was probably a bad idea. Will rewrite.